src/tokenizer.ts

```diff
@@ -160,14 +165,14 @@ function scanDollarQuotedString (state: State): Token {
   };
 }
 
-function scanString (state: State): Token {
+function scanString (state: State, dialect: Dialect): Token {
   let nextChar: Char;
-
+  const endString: Char[] = dialect === 'mysql' ? ['"', "'"] : ["'"]
   do {
     nextChar = read(state);
-  } while (nextChar !== '\'' && nextChar !== null);
+  } while (!endString.includes(nextChar) && nextChar !== null);
```

Comment:
Using an includes here breaks on mixing identifiers within the string, as it'll end up ending the string in the middle of a string. Consider the following query: `SELECT "foo'bar"` in MySQL. Under this code, we end up with the following tokenized sequence:
```js
{ type: 'keyword', value: 'SELECT', start: 0, end: 5 }
{ type: 'whitespace', value: ' ', start: 6, end: 6 }
{ type: 'string', value: '"foo\'', start: 7, end: 11 }
{ type: 'unknown', value: 'bar', start: 12, end: 14 }
{ type: 'string', value: '"', start: 15, end: 15 }
```

We can see here that the tokenizer has entered into a messed up state that if we were to use `SELECT "foo'bar"; SELECT * FROM table;`, we'll get:

```js
{ type: 'keyword', value: 'SELECT', start: 0, end: 5 }
{ type: 'whitespace', value: ' ', start: 6, end: 6 }
{ type: 'string', value: '"foo\'', start: 7, end: 11 }
{ type: 'unknown', value: 'bar', start: 12, end: 14 }
{ type: 'string',
  value: '"; SELECT * FROM table;',
  start: 15,
  end: 37 }
```

We need to capture the first character that starts the string, and then use it as the closing character.

Reply:
Ah, you're right, hmm.

Reply:
This isn't a class, so there's not a simple way to share state annoyingly. I'll think on it.

Might just push this for the ANSI operator first (")

Reply:
We don't need to share state though? I believe we can get away with doing a transformation from a do-while to something like:

```
let nextChar = read(state);
const openChar = nextChar;
while (nextChar !== openChar && nextChar !== null) {
  nextChar = read(state);
}
```

to enter the function, we know that the first `read(state)` will return a character from the `endString` array so I do not think we need to even bother testing against that array, rather just keep in memory what the first character is and test against that as we go to find the ending character.